spec/lib/services/sku-pack-service-spec.js

```diff
@@ -183,18 +187,154 @@ describe("SKU Pack Service", function() {
+                    }
+                ],
+                "discoveryGraphName": "Graph.InstallCoreOS",
+                "discoveryGraphOptions": {
+                    "username": "testuser",
+                    "password": "hello",
+                    "hostname": "mycoreos"
+                },
+                "createdAt": "2016-04-07T12:25:18.529Z",
+                "updatedAt": "2016-04-07T12:25:18.529Z",
+                "id": "570651ae87b3579d76508d26"
+            };
+            waterline.skus.updateByIdentifier.resolves({id: '570651ae87b3579d76508d26',
+                name: 'updatedSkuName'});
             skuService.regenerateSkus.resolves();
-            return skuService.postSku(sku).then(function(val){
-                expect(waterline.skus.create).to.have.been.called;
+            return skuService.patchSku(sku.id).then(function (val) {
+                expect(waterline.skus.updateByIdentifier).to.have.been.called;
+                expect(val.name).equal('updatedSkuName');
+            });
+        });
+    });
+
+    describe('should upsert a sku', function() {
+        before(function () {
+            sinon.stub(skuService, 'postSku');
+            sinon.stub(skuService, 'patchSku');
+
+        });
+        after(function () {
+            skuService.postSku.restore();
+            skuService.patchSku.restore();
+        });
+        var sku =
+        {
+            "name": "my test sku",
+            "rules": [
+                {
+                    "path": "dmi.Base Board Information.Manufacturer",
+                    "contains": "Intel"
+                },
+                {
+                    "path": "ohai.dmi.memory.total",
+                    "equals": "32946864kB"
+                }
+            ],
+            "discoveryGraphName": "Graph.InstallCoreOS",
+            "discoveryGraphOptions": {
+                "username": "testuser",
+                "password": "hello",
+                "hostname": "mycoreos"
+            },
+            "createdAt": "2016-04-07T12:25:18.529Z",
+            "updatedAt": "2016-04-07T12:25:18.529Z",
+            "id": "570651ae87b3579d76508d26"
+        };
+
+        it('should return sku if a sku is posted', function(){
+            skuService.postSku.resolves(sku);
+            return skuService.upsertSku(sku).then(function (val) {
                 expect(val.rules).equal(sku.rules);
                 expect(val.discoveryGraphName).equal(sku.discoveryGraphName);
                 expect(val.username).equal(sku.username);
                 expect(val.password).equal(sku.password);
                 expect(val.hostname).equal(sku.hostname);
             });
         });
+
+        it('should return err status if postSku fails a non-409', function(){
+            var err = new Errors.BaseError('internal server error');
+            err.status = 500; //Checking if not 409
+            return skuService.upsertSku(sku).catch(function (err) {
```

Comment:
You need some special handling if you want to put  some assertion logic in the `catch` block. Image the `skuService.upsertSku` is actually resolved, so the `catch` block will not be executed. Mocha will mark the empty resolved promise as passed.  Below is some code snippet:
```javascript
return skuService.upsertSku(sku)
.then(function() {
    throw new Error('upsertSku should be rejected!');
})
.catch(function (err) {
    expect(err.status).equal(500);
});
```